python/lsst/ip/isr/calibType.py

```diff
@@ -332,17 +335,22 @@ def readText(cls, filename, **kwargs):
             Raised if the filename does not end in ".ecsv" or ".yaml".
         """
         if filename.endswith((".ecsv", ".ECSV")):
-            data = Table.read(filename, format='ascii.ecsv')
-            return cls.fromTable([data], **kwargs)
-
+            data = Table.read(filename, format="ascii.ecsv")
+            calibClass = cls
+            if "CALIBCLS" in data.meta:
+                calibClass = doImport(data.meta["CALIBCLS"])
+            return calibClass.fromTable([data], **kwargs)
```

Comment:
It might be worth adding a:
```python
if calibClass is IsrCalib:
    raise ValueError(f"Calibration file {filename} does not know its calibration class and so can not be read using IsrCalib")
```
because otherwise you can going to get a random error from `fromTable` that will likely not explain the actual problem.

That might suggest we need a helper method

```python
@classmethod
def determine_class(cls, metadata: Dict[str, Any], msg: str = "") -> Type:
    calibClass_name = metadata.get("CALIBCLS")
    calibClass = doImport(calibClass_name) if calibClass_name is not None else cls
    if calibClass is IsrCalib:
        raise ValueError(f"Can not use base class to read old data: {msg}")
    return calibClass
```
(I'm exploring possibilities with those first two lines but what you have for those is also fine). The point though is that once we have that check for base class and old file format in place there becomes enough code duplication that it should be factored out.